app/controllers/profile_controller.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class ProfileController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_user
+
+  def edit_role; end
+
+  def customer_setup_info; end
+
+  def developer_setup_info; end
+
+  # This module smells of :reek:NilCheck
+  def update_role
+    role = params[:role]&.to_sym
+    raise "Unknown role: #{role}" unless User::USER_ROLES.key?(role)
+    @user.update(role: role)
```

Comment:
What if update fail?

Reply:
add if block and else redirect_to back to role choice